### DISCUSSION TITLE
"group by" field1, field2, and in each group, value of field3 must be different, then string_agg that field3 in each row

### DISCUSSION THREAD

**Discussion by fengchutadie** [duckdb/duckdb#13535] [Q&A]
![image](https://github.com/user-attachments/assets/5f445066-8aba-4d1e-abaa-a6cb61cc5372)
suppose I have a table in duckdb as the first table in picture, then I'd like to generate the second table in picture with some sql.

I should "group by" year and month, but I get a requirement that same value of "type" shouldn't be aggregated, different value of "type" should be aggregated, 
then as result, it's min(id), string_agg(type), year, month, any_value(paramA), any_value(paramB), and any_value(paramC)

Generally, we "group by" rows if value of some fields are the same in "group by" condition, as in example, they are "year" and "month".
but how to set restriction that value of other field ("type" field in this case) must be different in a group "group by" year and month,.

is it possible to implement it in duckdb by sql operation? such as 
select 
    min(id), string_agg(type), year, month, any_value(paramA), any_value(paramB), and any_value(paramC)
from table 
    group by year, month

how to add type condition in this sql then same type couldn't be aggregated in a group?

**Comment by soerenwolfers** [comment]
use `row_number() OVER (PARTITION BY year, month, type) AS row` first, then `GROUP BY year, month, row`

**Reply by fengchutadie** [reply]
yes, it works. Thanks
